sumpy/expansion/__init__.py

```diff
@@ -211,228 +226,280 @@ def get_full_kernel_derivatives_from_stored(self, stored_kernel_derivatives,
 
 # {{{ sparse matrix-vector multiplication
 
-def _spmv(spmat, x, sparse_vectors):
+class CSEMatVecOperator(object):
     """
-    :param spmat: maps row indices to list of (col idx, value)
-    :param x: maps vector indices to values
-    :param sparse_vectors: If True, treat vectors as dict-like, otherwise list-like
+    A class to facilitate a fast matrix vector multiplication with
+    common subexpression eliminated. In compressed Taylor
+    series, the compression matrix's operation count can be
+    reduced to `O(p**d)` from `O(p**{2d-1})` using CSE.
+
+    .. attribute:: assignments
+
+        A object of type List[Tuple[List[Tuple[int, Any]], List[Tuple[int, Any]]]].
```

Comment:
Want to make that an official type annotation (and enable the [sphinx ext](https://github.com/agronholm/sphinx-autodoc-typehints/) that incorporates them)?

Reply:
Does that work with `attribute`? Looks like it works only for `rtype` and `param`

Reply:
You're right... that's too bad. And it looks like sphinx just does that out of the box, so the extension isn't even needed.